pennylane/allocation.py

```diff
@@ -206,50 +201,67 @@ def __exit__(self, *_, **__):
+        num_wires (int): the number of wires to dynamically allocate.
 
     Keyword Args:
-        require_zeros (bool): whether or not the wires must start in the ``0`` state
-        restored (bool): whether or not the wires are returned to the same state they started in.
+        require_zeros (bool): 
+            specifies whether to allocate ``num_wires`` in the all-zeros state (``True``) or in any 
+            arbitrary state (``False``). The default value is ``require_zeros=True``.
+
+        restored (bool): 
+            whether or not the dynamically allocated wires are returned to the same state they started 
+            in. ``restored=True`` indicates that the user promises to restore the dynamically allocated 
+            memory to its original state before :func:`~.deallocate` is called. ``restored=False`` indicates 
+            that the user does not promise to restore the dynamically allocated memory before :func:`~.deallocate` 
+            is called. The default value is ``False``.
+
+    Returns:
+        DynamicRegister: an object, behaving similarly to ``Wires``, that represents the dynamically 
+        allocated memory.
+
+    .. note::
+        The ``allocate`` function should be used with :func:`~.deallocate` if it is not used as a context 
+        manager.
+
+    .. see-also::
+        :func:`~.deallocate`
 
     This function can be used as a context manager with automatic deallocation (preferred) or with manual
     deallocation via :func:`~.deallocate`.
 
     .. code-block:: python
 
-        @qml.qnode(qml.device('default.qubit', wires=("a", "b")))
         def c():
-            with qml.allocation.allocate(2, require_zeros=True, restored=False) as wires:
+            qml.H("a")
+            qml.H("b")
+
+            with qml.allocate(2, require_zeros=True, restored=False) as wires:
                 qml.CNOT(wires)
 
-            wires = qml.allocation.allocate(2, require_zeros=True, restored=False)
+            wires = qml.allocate(2, require_zeros=True, restored=False)
             qml.IsingXX(0.5, wires)
-            qml.allocation.deallocate(wires)
-
-            return qml.probs()
+            qml.deallocate(wires)
 
+            return qml.expval(qml.Z("a"))
 
     >>> print(qml.draw(c, level="user")())
-    <DynamicWire>: ─╭Allocate─╭●─────────────╭Deallocate─┤  Probs
-    <DynamicWire>: ─╰Allocate─╰X─────────────╰Deallocate─┤  Probs
-    <DynamicWire>: ─╭Allocate─╭IsingXX(0.50)─╭Deallocate─┤  Probs
-    <DynamicWire>: ─╰Allocate─╰IsingXX(0.50)─╰Deallocate─┤  Probs
+                a: ──H───────────────────────────────────┤  <Z>
+                b: ──H───────────────────────────────────┤  
+    <DynamicWire>: ─╭Allocate─╭●─────────────╭Deallocate─┤  
+    <DynamicWire>: ─╰Allocate─╰X─────────────╰Deallocate─┤  
+    <DynamicWire>: ─╭Allocate─╭IsingXX(0.50)─╭Deallocate─┤  
+    <DynamicWire>: ─╰Allocate─╰IsingXX(0.50)─╰Deallocate─┤  
     >>> print(qml.draw(c, level="device")())
```

Comment:
If I make `c` a qnode that uses `qml.device("default.qubit", wires=("a", "b"))` as the device, setting `level="device"` still causes the error I mentioned on slack.

Reply:
This was a hope that we would pull off device integration, but we have not.  Perhaps just delete for now?